daq/faucet_states_collector.py

```diff
@@ -56,11 +68,23 @@ def process_port_learn(self, dpid, port, mac, timestamp):
         mac_table[FaucetStatesCollector.TABLE_ENTRY_PORT_MAC_LEARNING_TS] = datetime.fromtimestamp(timestamp).strftime("%Y-%m-%d %H:%M%S")
 
     @save_states
-    def process_config_change(self, dpid, restart_type, timestamp):
+    def process_config_change(self, timestamp, dpid, restart_type):
         config_change_table = self.switch_states\
             .setdefault(dpid, {})
 
         config_change_table[FaucetStatesCollector.TABLE_ENTRY_LAST_RESTART_TYPE] = restart_type
         config_change_table[FaucetStatesCollector.TABLE_ENTRY_LAST_RESTART_TS] = datetime.fromtimestamp(timestamp).strftime("%Y-%m-%d %H:%M:%S")
         config_change_table[FaucetStatesCollector.TABLE_ENTRY_CONFIG_CHANGE_COUNT] =\
             config_change_table.setdefault(FaucetStatesCollector.TABLE_ENTRY_CONFIG_CHANGE_COUNT, 0) + 1
+
+    @save_states
+    def process_stack_topo_change(self, timestamp, stack_root, graph):
+        topo_change_obj = self.topo_state\
+                .setdefault(FaucetStatesCollector.TOPOLOGY_ROOT, {})\
+                .setdefault(FaucetStatesCollector.TOPOLOGY_GRAPH, {})\
+
+        topo_change_obj[FaucetStatesCollector.TOPOLOGY_ROOT] = stack_root
+        topo_change_obj[FaucetStatesCollector.TOPOLOGY_GRAPH] = json.loads(graph)
```

Comment:
@grafnu I am converting graph object to JSON obj from string here. Passing an obj directly throws a "not serializable" error.